packages/features/bookings/Booker/components/hooks/useInitialFormValues.ts

```diff
@@ -124,9 +166,12 @@ export function useInitialFormValues({
         name: defaultUserValues.name,
         email: defaultUserValues.email ?? "",
       };
-      setDefaultValues(defaults);
+      setInitialValuesState({
+        values: defaults,
+        key: buildKey({ values: defaults, hasSession, stableHashExtraOptions }),
+      });
     })();
-    // do not add extraOptions as a dependency, it will cause infinite loop
+    // TODO: Remove it. It was initially added so that we don't add extraOptions in deps but that is handled using stableHashExtraOptions now.
```

Comment:
cc @ThyMinimalDev Added extraOptions as a dependency indirectly by using a stableHash of the content.

It was required because with AppRouter somehow the state isn't reset fully for booker on router.push and query params that are changing were being ignored by this specific code.